lib/charms/opensearch/v0/opensearch_snapshots.py

```diff
@@ -0,0 +1,1197 @@
+            return
+
+        # start the restore
+        self.charm.status.set(MaintenanceStatus(RestoreInProgress))
+        logger.info("Starting restore of snapshot %s.", snapshot_id)
+        try:
+            non_restored_indices = self.charm.snapshots_manager.restore_snapshot(
+                object_storage_type=self.object_storage_type, snapshot=snapshot
+            )
+            if not non_restored_indices:
+                self.charm.health.apply(wait_for_green_first=True, app=self.charm.unit.is_leader())
+                return
+
+            logger.error(
+                "Failed to restore the following indices in snapshot %s: %s.",
+                snapshot_id,
+                non_restored_indices,
+            )
+            event.fail(
+                f"Failed to restore {len(non_restored_indices)} indices. Check logs for details."
+            )
+        except OpenSearchHttpError as e:
+            logger.error("Failed to restore snapshot %s. Error: %s.", snapshot_id, str(e))
+            event.fail(f"Failed to restore snapshot {snapshot_id}. Error: {str(e)}.")
+        finally:
+            self.charm.status.clear(RestoreInProgress)
+
+    def _on_peer_clusters_relation_changed_for_snapshots(self, event):  # noqa C901
+        """Apply snapshots config when the orchestrator broadcasts over peer-clusters."""
+        # Only leaders perform cluster-level config
+        if not self.charm.unit.is_leader():
+            return
+
+        dep = self.charm.opensearch_peer_cm.deployment_desc()
+        if not dep:
+            event.defer()
+            return
+
+        # When it is the main orchestrator, it is same with single cluster
+        # so we don't need to apply cluster-level config.
+        if dep.typ == DeploymentType.MAIN_ORCHESTRATOR:
+            return
+
+        # read the effective storage type/config coming from peer-clusters
+        os_type = self.charm.peers_data.get(Scope.APP, OS_PEER_KEY_TYPE)
+        repo = self.charm.peers_data.get_object(Scope.APP, OS_PEER_KEY_REPO)
+        if not os_type or not repo or not repo.get("secret_id"):
+            return
+
+        secret_id = repo.get("secret_id")
+        secret_payload = {}
+        if secret_id:
+            try:
+                secret = self.charm.model.get_secret(id=secret_id)
+                secret_payload = secret.get_content()
+            except Exception as e:
+                logger.error("Failed to read object-storage secret %s: %s", secret_id, e)
+                event.defer()
+                return
+
+        if repo and repo.get("type") == "s3":
+            self.charm.snapshots_manager.store_s3_ca(repo.get("tls_ca_chain") or None)
+            self.charm.keystore_manager.put_entries(
+                {
+                    "s3.client.default.access_key": secret_payload.get("access-key", ""),
+                    "s3.client.default.secret_key": secret_payload.grt("secret-key", ""),
```

Comment:
```suggestion
                    "s3.client.default.secret_key": secret_payload.get("secret-key", ""),
```

Empty keys should not be added to the keystore imo

Reply:
This is fixed.